src/smc-webapp/share/page.cjsx

```diff
@@ -123,21 +133,23 @@ TopBar = ({viewer, path, project_id, base_url, site_name}) ->
             v = v.slice(0, v.length-1)
         segments = []
         t = ''
+
         v.reverse()
-        for s in v
-            href = "#{t}?viewer=share"
-            if t
-                segments.push(<a key={t} href={href}>{s}</a>)
+        for val, i in v
+            path_of_segment = v.slice(i).reverse().join('/')
+            if t and misc.path_is_in_public_paths(path_of_segment, list_of_paths.toJS())
+                href = "#{t}?viewer=share"
+                segments.push(<a key={t} href={href}>{val}</a>)
             else
-                segments.push(<span key={t}>{s}</span>)
+                segments.push(<span key={t}>{val}</span>)
             if not t
                 if path.slice(-1) == '/'
                     t = '..'
                 else
                     t = '.'
             else
                 t += '/..'
-        segments.reverse()
+        segments.reverse()g
```

Comment:
You've introduced a syntax error (I assume)  here with that g that got accidentally put at the end of the line?

Reply:
Yeah, typo.